packages/cache/src/CacheStore.ts

```diff
@@ -95,6 +95,19 @@ export class CacheStore {
     this.syncSubscribers(options.cacheKey)
   }
 
+  public remove = (options?: Pick<CacheOptions<unknown, CacheKey>, 'cacheKey'>) => {
+    if (typeof options?.cacheKey === 'undefined' || options.cacheKey.length === 0) {
+      this.cacheStore.clear()
+      return
+    }
+
+    const hashedCacheKey = hashCacheKey(options.cacheKey)
+
+    if (this.cacheStore.has(hashedCacheKey)) {
+      this.cacheStore.delete(hashedCacheKey)
+    }
+  }
+
```

Comment:
How about changing the interface to removeCaches? Similar to removeQueries in React Query, the implementation currently deletes everything if no key is provided.
```suggestion
  public removeCaches = (options?: Pick<CacheOptions<unknown, CacheKey>, 'cacheKey'>) => {
    if (typeof options?.cacheKey === 'undefined' || options.cacheKey.length === 0) {
      this.cacheStore.clear()
      return
    }

    const hashedCacheKey = hashCacheKey(options.cacheKey)

    if (this.cacheStore.has(hashedCacheKey)) {
      this.cacheStore.delete(hashedCacheKey)
    }
  }

```

Reply:
Then how about remove one by one first?

```suggestion
  public remove = (options: Pick<CacheOptions<unknown, CacheKey>, 'cacheKey'>) => {
    const hashedCacheKey = hashCacheKey(options.cacheKey)

    if (this.cacheStore.has(hashedCacheKey)) {
      this.cacheStore.delete(hashedCacheKey)
    }
  }

```

### Suggestion (now)

cacheStore.remove(errorCache(1))
cacheStore.remove() // typescript type error

### Alternatives 1 (later)

cacheStore.removeAll() 

### Alternatives 2 (later)

cacheStore.remove() // remove cache all

Reply:
Cool! Currently, the reset method is implemented such that if no key is provided, it resets everything. Should we change this implementation as well?